app/src/main/java/com/infomaniak/mail/ui/login/IntroFragment.kt

```diff
@@ -109,29 +106,21 @@ class IntroFragment : Fragment() {
         lifecycleScope.launch(Dispatchers.IO) {
             val duration = resources.getInteger(R.integer.loginLayoutAnimationDuration).toLong()
             delay(duration)
-            introViewModel.currentAccentColor.postValue(accentColor)
+            loginViewModel.currentAccentColor.postValue(accentColor)
         }
     }
 
     private fun updateUiWhenThemeChanges(position: Int?) {
-        introViewModel.currentAccentColor.observe(viewLifecycleOwner) { accentColor ->
-            setUi(
-                accentColor = accentColor ?: UiSettings.DEFAULT_ACCENT_COLOR,
-                position = position,
-                animate = accentColor != null,
-            )
+        loginViewModel.currentAccentColor.observeNotNull(viewLifecycleOwner) { accentColor ->
+            setUi(accentColor, position)
         }
     }
 
-    /**
-     * `animate` is necessary because when the activity is started we want to avoid animating the color change the first time.
-     */
-    private fun setUi(accentColor: AccentColor, position: Int?, animate: Boolean = true) = with(binding) {
+    private fun setUi(accentColor: AccentColor, position: Int?) = with(binding) {
+        val animate = position == 0
         updateEachPageUi(accentColor, animate)
-        if (position == 0) {
-            updateFirstPageUi(accentColor, animate)
-            if (navigationArgs.isFirstAccount) updateActivityUi(accentColor, animate)
-        }
+        if (position == 0) updateFirstPageUi(accentColor, animate)
+        if (position == 3) updateActivityUi(accentColor, animate)
```

Comment:
Add comment to explain why doing it only at position == 3 is important

Reply:
At position 0, animate is always true
At position 3, animate is always false
Please review this code

Reply:
Magic numbers have been removed.

There was also a bug that was fixed, so `animate` computation has been reworked.